superduperdb/backends/sqlalchemy/metadata.py

```diff
@@ -392,3 +407,29 @@ def get_model_queries(self, model: str):
                     {'query_id': id, 'query': query, 'sql': query.repr_()}
                 )
             return unpacked_queries
+
+
+def convert_object_id_type(job: Job):
+    kwargs = job.kwargs
+    ids = kwargs.get('ids', [])
+    if not isinstance(ids, list):
+        return
+
+    new_ids = []
+    for id in ids:
+        if isinstance(id, ObjectId):
+            new_ids.append(str(id))
+    kwargs['ids'] = new_ids
+
+
+def recovery_object_id_type(job):
+    kwargs = job.kwargs
+    ids = kwargs.get('ids', [])
+    if not isinstance(ids, list):
+        return
+
+    new_ids = []
+    for id in ids:
+        if isinstance(id, str):
+            new_ids.append(ObjectId(id))
+    kwargs['ids'] = new_ids
```

Comment:
If we use MongoDB for data_backend and use SQL for metadata_store. metadata_store will get errors due to the `ObjectId(id)` from data MongoDB